task_spec/spec/acceptance/init_spec.rb

```diff
@@ -100,19 +104,44 @@ def target_platform
       expect(res['result']['source']).to be
     end
 
-    # Upgrade from puppet5 to puppet6
+    # Puppet Agent can't be upgraded on Windows nodes while 'puppet agent' service or 'pxp-agent' service are running
+    if target_platform =~ /win/
+      # Try to upgrade from puppet5 to puppet6 but fail due to puppet agent service already running
+      results = run_task('puppet_agent::install', 'target', { 'collection' => 'puppet6', 'version' => 'latest' })
+      results.each do |res|
+        expect(res).to include('status' => 'failure')
+        expect(res['result']['_error']['msg']).to match(%r{Puppet Agent upgrade cannot be done while Puppet services are still running.})
+      end
+
+      # Manually stop the puppet agent service
+      service = run_command('c:/"program files"/"puppet labs"/puppet/bin/puppet resource service puppet ensure=stopped', 'target')
+      output = service[0]['result']['stdout']
+      expect(output).to include("ensure => 'stopped'")       
+    end
+
+    # Succesfully upgrade from puppet5 to puppet6
     results = run_task('puppet_agent::install', 'target', { 'collection' => 'puppet6', 'version' => 'latest' })
     results.each do |res|
       expect(res).to include('status' => 'success')
     end
 
     # Verify that it upgraded
+    installed_version = nil
```

Comment:
What is this for?

Reply:
Why is this set to `nil`?

Reply:
It's there to simply avoid an `undefined local variable or method` error. Its role is to store the currently installed puppet agent version (since we've previously installed the latest puppet6 version) and it's used below, in the next step, when trying to install the same version again. 

I have no doubt that there are better/more elegant approaches to testing this. I just tried to be minimalist with the changes and use the existing steps as much as possible. So, if there are any concerns, I'm always open to improvement ideas 😃.

Reply:
I see. I did not see that was used below.